pkg/controller/cronjob/utils_test.go

```diff
@@ -426,8 +421,8 @@ func TestGetLatestUnmetScheduleTimes(t *testing.T) {
 		// Deadline is short
 		deadline := int64(2 * 60 * 60)
 		cj.Spec.StartingDeadlineSeconds = &deadline
-		times := getUnmetScheduleTimes(cj, now, PraseSchedule(cj.Spec.Schedule), recorder)
-		if len(times) == 0 {
+		schedule := getNextScheduleTime(cj, now, PraseSchedule(cj.Spec.Schedule), recorder)
+		if schedule == nil {
```

Comment:
These entire test case are begging to be re-written as table tests.

Reply:
+1, will try to take it as a follow-up PR